modules/core/src/test/java/org/apache/ignite/platform/BinaryContextPlatformDotNetSessionLockResultPredefinedTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.platform;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.binary.BinaryClassDescriptor;
+import org.apache.ignite.internal.binary.BinaryContext;
+import org.apache.ignite.internal.binary.TestCachingMetadataHandler;
+import org.apache.ignite.internal.processors.platform.websession.PlatformDotNetSessionLockResult;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests for registering PlatformDotNetSessionLockResult class.
+ */
+public class BinaryContextPlatformDotNetSessionLockResultPredefinedTest {
```

Comment:
Can we also add a generic test that no constructors are created during default node startup and that ReflectionFactory is not used? In this case the test may be renamed to BinaryContextConstructorGenerationTest perhaps.

Can you also add a custom class with no public empty constructor, add it to binary context, and make sure that for it we generate two distict constructors as per ReflectionFactory?